File: backend_java/src/main/java/br/com/fiap/chronosdtn/model/Account.java
package br.com.fiap.chronosdtn.model;

// Por que: Importa as anotações do Jakarta Persistence para persistência das contas monetárias.
import jakarta.persistence.*;
// Por que: Importa as anotações do Lombok para reduzir código boilerplate de get/set.
import lombok.*;
// Por que: Importa BigDecimal para garantir precisão absoluta nas operações de saldo e câmbio.
import java.math.BigDecimal;

// Por que: Define a classe como entidade JPA.
@Entity
// Por que: Vincula a classe à tabela física de contas operacionais.
@Table(name = "TB_CHRONOS_ACCOUNT")
// Por que: Lombok - Getters para todos os campos.
@Getter
// Por que: Lombok - Setters para todos os campos.
@Setter
// Por que: Lombok - Construtor padrão com todos os campos de conta.
@AllArgsConstructor
// Por que: Lombok - Construtor vazio.
@NoArgsConstructor
public class Account {

    // Por que: Identificador único da conta bancária.
    @Id
    @Column(name = "ID_ACCOUNT")
    private Long id;

    // Por que: Relacionamento de muitos-para-um vinculando a conta ao nó que a gerencia.
    @ManyToOne(fetch = FetchType.EAGER)
    // Por que: Chave estrangeira que aponta para o ID do nó hospedeiro.
    @JoinColumn(name = "ID_NODE", nullable = false)
    private Node node;

    // Por que: Nome da empresa ou entidade titular da conta espacial.
    @Column(name = "NM_HOLDER", nullable = false, length = 100)
    private String holder;

    // Por que: Saldo financeiro, usando BigDecimal para evitar perda de precisão binária de ponto flutuante.
    @Column(name = "VL_BALANCE", nullable = false, precision = 15, scale = 2)
    private BigDecimal balance;

    // Por que: Nome da moeda da conta (ex: USD, LUN).
    @Column(name = "NM_CURRENCY", nullable = false, length = 10)
    private String currency;
}

File: backend_net/ChronosDtnNetApi/Models/Account.cs
namespace ChronosDtnNetApi.Models;

// Por que: Representa a conta monetária de uma corporação ou mineradora espacial hospedada em um nó físico.
public class Account
{
    // Por que: Código numérico único identificador da conta bancária.
    public long Id { get; set; }

    // Por que: ID do nó onde esta conta é liquidada localmente (ex: 1 para Terra, 2 para Lua).
    public long NodeId { get; set; }

    // Por que: Instância do nó hospedeiro da conta para controle estrutural do banco.
    public Node? Node { get; set; }

    // Por que: Nome da empresa ou entidade titular da conta (ex: SpaceX Account).
    public string Holder { get; set; } = string.Empty;

    // Por que: Saldo financeiro com precisão adequada para operações de débito/crédito.
    public decimal Balance { get; set; }

    // Por que: Abreviação monetária da conta (ex: USD para dólar, LUN para moedas lunares).
    public string Currency { get; set; } = string.Empty;
}
